pkg/agent/agent.go

```diff
@@ -231,14 +226,18 @@ func (a *Agent) bootstrap() error {
 		a.baseSVIDKey = key
 
 		// If we're here, we need to attest/Re-attest
-		regEntryMap, err := a.attest()
-		if err != nil {
-			return err
-		}
-		err = a.FetchSVID(regEntryMap, a.BaseSVID, a.baseSVIDKey)
+		regEntries, err := a.attest()
 		if err != nil {
 			return err
 		}
+		uri := a.config.TrustDomain
+		uri.Path = "spiffe/cp"
+		cacheManager := manager.NewManager(
+			a.Cache, a.serverCerts,
+			uri.String(), a.config.ServerAddress.String(),
+			a.config.ErrorCh, a.BaseSVID, a.baseSVIDKey, regEntries, a.config.Log)
+		go cacheManager.UpdateCache(a.stop)
```

Comment:
@boz had once recommended not passing channels around (particularly between packages), and after having played around with them a bit I tend to agree. Here's the pattern I used the last time I ran into this: https://github.com/spiffe/spire/blob/18eb8142e9ea309d186bb09758e2881fe278943c/pkg/server/endpoint/endpoint.go#L178-L210

And then on the consuming side:
https://github.com/spiffe/spire/blob/18eb8142e9ea309d186bb09758e2881fe278943c/pkg/server/server.go#L153

I have found other common libraries to have similar block-forever behavior (like net/http server and grpc server)

Reply:
Thank you @boz for explaining it so well. I will make the necessary changes to use ```context.Context ``` instead of a stop channel and implement the decoupling pattern you explain here.

Reply:
Welp, looks like I stand corrected :) blocking forever can be dangerous - I just hit this one in the very example I linked previously https://github.com/golang/go/issues/20239